### PR TITLE
Add link to documentation in tsconfig.json template generated by tsc --init 

### LINKED ISSUE #34538: Add Advanced Compiler flags in tsconfig.json generated by --init
## Search Terms

--init
compileroptions
advanced compiler options
resolveJsonModule
tsconfig
default tsconfig

## Suggestion

I assumed that the `tsconfig.json` that is generated by `tsc --init` provides all available flags with their defaults.

I noticed now that there is a mismatch of the available fields documented here: https://www.typescriptlang.org/docs/handbook/compiler-options.html and the generated tsconfig.json file.

For example the option 'resolveJsonModule' is missing. However there are actually a long list of flags missing. These are all currently categorized as "Advanced".

### Option 1: Just add the missing flags to the generated tsconfig.
There is already a category "Experimental" Features. I do not think it would confuse a new TypeScript developer if there is an additional category "Advanced" generated in the 'tsconfig.json' file.

### Option 2: have two different init commands
* tsc --init -> creates the current tsconfig.json
* tsc --initAdv -> creates a full tsconfig.json

(Not really a good solution, and makes things more complicated)

### Option 3: Add a comment
Add a comment to the currently generated `tsconfig.json`
something like: "For more Information, and Advanced flags visit: https://www.typescriptlang.org/docs/handbook/compiler-options.html"

## Further information
I assumed first it was a bug, that it was forgotten to be added to the default config. However looking at the codebase I discovered:
in the file: 
.\src\compiler\commandLineParser.ts

Line 1973: // Exclude all possible `Advanced_Options` in tsconfig.json which were NOT defined in command line

I could not find any more information why this decision has been made. - But since all information is already available the implementation of this 'feature' should be trivial.

## Use Cases

I would like as a user of typescript to have a complete overview of all available compiler flags.
It is akward to have a default template that forgets to mention that even more flags exist.

It would save people time to search for issues that are easily solved by a compile flag.

## Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

### PR DESCRIPTION
**Pull request by borgfriend**
Fixes #34538

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
A PR adding the link to the top is OK, but again, we do not want literally every compiler flag listed in the init tsconfig. For example, some of the flags listed here *do nothing*, are *explicitly discouraged*, or have *dangerous effects*.

**Comment by DanielRosenwasser** [comment]
Reopening in case you want to just make the PR about providing the link to other options. In that case, I would also suggest we remove the comment descriptions because they tend to be a major pain when using `tsc --init`.

**Comment by borgfriend** [comment]
I removed the Advanced Options again... however I would suggest to move various options then into better category names like "Deprecated", "Use with Caution" etc.

Me not being really familiar with the codebase just assumed these are all "Advanced options" that you should be able to use as an advanced user.

In addition I removed the comment descriptions as you suggested.

**Comment by sandersn** [comment]
@borgfriend Can you update the description of this PR? I'm triaging our old PRs that we forgot about and I can't tell what this evolved into, but it seems different than the initial description.

**Comment by borgfriend** [comment]
I think it is quite misleading to only expose a subset of flags available in tsconfig.json when generating it via tsc --init. 

The current flags are "handpicked" because - there are dangerous, deprecated flags that should not be exposed.

The issue evolved into:
- add link to documentation
- remove documentation strings 

Thus users would be encouraged to take a look at the documentation to learn more about available flags (also those that are not available in the 'handpicked selection')

**Comment by sheetalkamat** [comment]
@borgfriend  can you please update this to resolve merge conflicts.. Thanks

**Comment by orta** [comment]
There's an example of what we'd like it to look like in https://github.com/microsoft/TypeScript/issues/37361#issuecomment-598867169 too

**Comment by borgfriend** [comment]
@sheetalkamat I fixed the merge conflicts, @orta adjusted the link to be like in the example

File: src/compiler/commandLineParser.ts [hunk 1]
```diff
@@ -150,6 +150,14 @@ namespace ts {
             category: Diagnostics.Command_line_Options,
             description: Diagnostics.Whether_to_keep_outdated_console_output_in_watch_mode_instead_of_clearing_the_screen,
         },
+        {
```
Comment (sheetalkamat) [inline src/compiler/commandLineParser.ts]: There is already incremental included in this list why is there another entry?

File: tests/baselines/reference/tsConfig/Default initialized TSConfig/tsconfig.json [hunk 2]
```diff
@@ -1,66 +1,69 @@
 {
   "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
     /* Basic Options */
-    // "incremental": true,                   /* Enable incremental compilation */
```
Comment (sheetalkamat) [inline tests/baselines/reference/tsConfig/Default initialized TSConfig/tsconfig.json]: Example shows the comments as not removed https://github.com/microsoft/TypeScript/issues/37361#issuecomment-598867169 @RyanCavanaugh @DanielRosenwasser Whats the final call on that

File: src/compiler/commandLineParser.ts (hunk 1 above)
Comment (borgfriend) [reply src/compiler/commandLineParser.ts]: guess that was automerged and I missed it - pushed a fix for it

Comment (sheetalkamat) [reply src/compiler/commandLineParser.ts]: Can you please existing entry instead? Is there reason this needs to move ?

Comment (borgfriend) [reply src/compiler/commandLineParser.ts]: better now?

File: src/compiler/commandLineParser.ts [hunk 3]
```diff
@@ -2108,10 +2108,9 @@ namespace ts {
 
         function isAllowedOption({ category, name }: CommandLineOption): boolean {
             // Skip options which do not have a category or have category `Command_line_Options`
-            // Exclude all possible `Advanced_Options` in tsconfig.json which were NOT defined in command line
```
Comment (DanielRosenwasser) [inline src/compiler/commandLineParser.ts]: Return the comment, remove the extra `;`

Comment (borgfriend) [reply src/compiler/commandLineParser.ts]: done

File: tests/baselines/reference/tsConfig/Default initialized TSConfig/tsconfig.json [hunk 4]
```diff
@@ -1,66 +1,68 @@
...
-    /* Strict Type-Checking Options */
-    "strict": true,                           /* Enable all strict type-checking options. */
-    // "noImplicitAny": true,                 /* Raise error on expressions and declarations with an implied 'any' type. */
-    // "strictNullChecks": true,              /* Enable strict null checks. */
-    // "strictFunctionTypes": true,           /* Enable strict checking of function types. */
-    // "strictBindCallApply": true,           /* Enable strict 'bind', 'call', and 'apply' methods on functions. */
-    // "strictPropertyInitialization": true,  /* Enable strict checking of property initialization in classes. */
-    // "noImplicitThis": true,                /* Raise error on 'this' expressions with an implied 'any' type. */
-    // "alwaysStrict": true,                  /* Parse in strict mode and emit "use strict" for each source file. */
+    /* Basic Options */                       /* Enable incremental compilation */
```
Comment (sheetalkamat) [inline tests/baselines/reference/tsConfig/Default initialized TSConfig/tsconfig.json]: The comments are off by a line ?

Comment (borgfriend) [reply tests/baselines/reference/tsConfig/Default initialized TSConfig/tsconfig.json]: This is rather odd, i just reverted my change of removing them.
I adjusted the code so an offset is not possible anymore

**Comment by borgfriend** [comment]
I just noticed that for the flag "target" the description states the default is 'es3' - however it is set to 'es5'? 

Maybe the target and description should be updated to ES2017 (version supported by node 8, and all current Browsers without IE)?

Should I change the flag?

File: src/compiler/commandLineParser.ts (hunk 3 above)
Comment (sheetalkamat) [reply src/compiler/commandLineParser.ts]: The comment is still gone. Can you please fix that so that its ready to merge

**Comment by sheetalkamat** [comment]
@borgfriend default for target is es3 (that means if target is not specified it is es3). The comment is correct

**Pull request merged**